presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -114,6 +115,11 @@ public ProjectNode project(Assignments assignments, PlanNode source)
         return new ProjectNode(idAllocator.getNextId(), source, assignments);
     }
 
+    public MarkDistinctNode markDistinct(Symbol markerSymbol, List<Symbol> distinctSymbols, Optional<Symbol> hashSymbol, PlanNode source)
```

Comment:
What do you think about having two methods, one with hashSymbol and one without? That way you would need to pass Optional (at least into the API of `PlanBuilder`)